package/sbp_nmea2000_bridge/src/sbp_nmea2000_bridge.cpp

```diff
@@ -0,0 +1,917 @@
+                                            &seconds_since_midnight);
+      d << "\tTOW: " << sbp_utc_time->tow << "\n"
+        << "\tDays since 1970: " << days_since_1970 << "\n"
+        << "\tSeconds since midnight: " << seconds_since_midnight << "\n";
+
+      bool is_valid = is_flags_valid(sbp_utc_time->flags, /*bits=*/3);
+
+      tN2kMsg N2kMsg;
+      if (is_valid) {
+        SetN2kSystemTime(N2kMsg, tow_sequence_id_cache[sbp_utc_time->tow],
+                         days_since_1970, seconds_since_midnight,
+                         tN2kTimeSource::N2ktimes_GPS);
+      } else {
+        SetN2kSystemTime(N2kMsg, /*sequence ID=*/0xFF,
+                         N2kUInt16NA, N2kDoubleNA,
+                         tN2kTimeSource::N2ktimes_GPS);
+      }
+      NMEA2000.SendMsg(N2kMsg);
+
+      n2k_composite_message_cache.ResetIfOld(sbp_utc_time->tow);
+      n2k_composite_message_cache.SetFields(sbp_utc_time->tow,
+                                            days_since_1970,
+                                            seconds_since_midnight);
+      if (is_valid &&
+          n2k_composite_message_cache.IsReadyToSend(/*is_valid=*/true)) {
+        N2kMsg.Clear();
+        n2k_composite_message_cache.FillN2kMsg(&N2kMsg, /*is_valid=*/true);
+        NMEA2000.SendMsg(N2kMsg);
+      }
+
+      d << "\tDone.\n";
+    }
+
+    // PGN 127250
+    void callback_sbp_baseline_heading(u16 sender_id, u8 len, u8 msg[], void *context){
+      UNUSED(sender_id);
+      UNUSED(len);
+      UNUSED(context);
+      d << __FUNCTION__ << "\n";
+
+      auto sbp_baseline_heading =
+              reinterpret_cast<msg_baseline_heading_t*>(msg);
+
+      d << "\tBaseline heading in deg: "
+        << sbp_baseline_heading->heading / 1000.0 << "\n"
+        << "\tBaseline heading in rad: "
+        << DegToRad(sbp_baseline_heading->heading / 1000.0) << "\n";
+
+      tN2kMsg N2kMsg;
+      SetN2kTrueHeading(N2kMsg, tow_sequence_id_cache[sbp_baseline_heading->tow],
+                        DegToRad(sbp_baseline_heading->heading / 1000.0));
+      NMEA2000.SendMsg(N2kMsg);
+
+      d << "\tDone.\n";
+    }
+
+    // PGN 129025
+    void callback_sbp_pos_llh(u16 sender_id, u8 len, u8 msg[], void *context) {
+      UNUSED(sender_id);
+      UNUSED(len);
+      UNUSED(context);
+      d << __FUNCTION__ << "\n";
+
+      auto *sbp_pos = reinterpret_cast<msg_pos_llh_t*>(msg);
+
+      d << "\tTOW: " << sbp_pos->tow << "\n"
+        << "\tLat: " << sbp_pos->lat << "\n"
+        << "\tLon: " << sbp_pos->lon << "\n"
+        << "\tAlt: " << sbp_pos->height << "\n";
+
+      bool is_valid =is_flags_valid(sbp_pos->flags, /*bits=*/3);
+
+      tN2kMsg N2kMsg;
+      if(is_valid) {
+        SetN2kLatLonRapid(N2kMsg, sbp_pos->lat, sbp_pos->lon);
+      } else {
+        SetN2kLatLonRapid(N2kMsg, N2kDoubleNA, N2kDoubleNA);
+      }
+      NMEA2000.SendMsg(N2kMsg);
+
+      // In case of invalid position, pos_llh message marks an epoch.
```

Comment:
If I understand correctly, reworded for clarity: "If the value of pos_llh is invalid, it marks an epoch, therefore..."